src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -4540,9 +4540,15 @@ internal bool TryReadSqlValue(SqlBuffer value, SqlMetaDataPriv md, int length, T
 
         private bool TryReadSqlDateTime(SqlBuffer value, byte tdsType, int length, byte scale, TdsParserStateObject stateObj)
         {
+#if netcoreapp
+            Span<byte> datetimeBuffer = length <= 10 ? stackalloc byte[length] : new byte[length];
```

Comment:
why not just the span version ?

Reply:
System.Memory is only included in specific builds.